change-notes/1.21/analysis-javascript.md

```diff
@@ -5,6 +5,7 @@
 * Support for the following frameworks and libraries has been improved:
   - [koa](https://github.com/koajs/koa)
   - [socket.io](http://socket.io)
+  - [shelljs](https://www.npmjs.com/package/shelljs)
```

Comment:
```suggestion
  - [shelljs](https://github.com/shelljs/shelljs)
```

Reply:
Why are we linking to github repos instead of the NPM page, again?

Reply:
> GitHub repo > npm page ?

I would say `<`

Reply:
The github repository is the official definition of the package, in the absence of a proper home page (like https://socket.io has). The npm page is just derived from the repository information, and our models are not tied to npm in any way.

Reply:
The npm page looks much more useful to me, and it has a link to the github repo.

Reply:
But this is not a matter of usefulness: it is a matter of documenting what we are doing, not how to use a package. I am fine either way, as long as we are consistent.
Should the preference be `homepage > npm > repo`?

Reply:
Yes, I would prefer that preference order.

Reply:
The NPM page is also the more stable link, as it's not unusual for github repos to be renamed or moved into an org as the project grows.